docs(api): restructure

mffap opened the pull request:
### Definition of Ready

- [ ] I am happy with the code
- [ ] Short description of the feature/issue is added in the pr description
- [ ] PR is linked to the corresponding user story
- [ ] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [ ] No debug or dead code
- [ ] My code has no repetitions
- [ ] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [ ] Documentation/examples are up-to-date
- [ ] All non-functional requirements are met
- [ ] Functionality of the acceptance criteria is checked manually on the dev system.

mffap commented:
It would be nice if we could automatically list the items under "Core resources", but I couldn't manage to do it. Maybe @peintnermax has an idea?

mffap commented:
@hifabienne you mentioned also the structure on the SDKs. Please make a suggestion.

Futhermore: The sidebar "Sign in users" could also contain the new resource based endpoints. It's a bit inconsistent now. Maybe you have an idea how to resolve.

mffap commented:
@dakshitha maybe you also have some input, feel free to add

hifabienne commented:
> @hifabienne you mentioned also the structure on the SDKs. Please make a suggestion.
> 
> Futhermore: The sidebar "Sign in users" could also contain the new resource based endpoints. It's a bit inconsistent now. Maybe you have an idea how to resolve.

I thought we didn't include community sdks in the list, but we already did. I just added the missing ones.

In the sign in users section I added a link to custom login, which redirects to the build your own login guides. I think it makes more sense to redirect to the guides as to the apis itself, as there you do have the whole flows that are needed, and not only the separat requests. What do you think about that?

mffap commented:
> In the sign in users section I added a link to custom login, which redirects to the build your own login guides. I think it makes more sense to redirect to the guides as to the apis itself, as there you do have the whole flows that are needed, and not only the separat requests. What do you think about that?

Thanks! Yes, good idea - I forgot to reference the guide. I moved it inline (custom authentication & authorization), since clicking on the link in the sidebar switches context from the API docs to the guides. 

I feel like the section could actually need a whole intro page, but we basically write it here now. I think this is sufficient for the moment as it explains the order and the different sections in the sidebar.
